src/background/vvppManager.ts

```diff
@@ -138,22 +145,99 @@ export class VvppManager {
         }
         return parseInt(aMatch[1]) - parseInt(bMatch[1]);
       });
-      for (const p of filePaths) {
-        streams.push(fs.createReadStream(p));
+      format = await this.detectFileFormat(filePaths[0]);
+      if (!format) {
+        throw new Error(`Unknown file format: ${filePaths[0]}`);
       }
+      files = filePaths;
     } else {
       log.log("Not a split file");
-      streams.push(fs.createReadStream(vvppLikeFilePath));
+      format = await this.detectFileFormat(vvppLikeFilePath);
+      if (!format) {
+        throw new Error(`Unknown file format: ${vvppLikeFilePath}`);
+      }
+      files = [vvppLikeFilePath];
     }
 
+    log.log("Format:", format);
     log.log("Extracting vvpp to", outputDir);
     try {
-      await new Promise((resolve, reject) => {
-        new MultiStream(streams)
-          .pipe(Extract({ path: outputDir }))
-          .on("close", resolve)
-          .on("error", reject);
-      });
+      let tmpFile: string | undefined;
+      let file: string;
```

Comment:
これもめっちゃ細かいのですが、この状態変数が何を示すのか、代入されるのがあとなのでここでわからない感じでした。
（コードを読む人はこの変数の意味がわからないままコードを読むことになるのでちょっとだけ可読性が落ちます）

もうちょっと意味のわかる変数名にするのはどうでしょう。例えばこんな感じ･･･？
```suggestion
      let concatenatedTmpFile: string | undefined;
      let archiveFile: string;
```